brave/browser-laptop
# Fixes publisher toggle (heart), button enables after time requirements satisfied.

Fixes #12675

## Submitter Checklist:


## Test Plan:
1. Enable payments
2. Select a minimum browsing time and visits in payment settings, (1 visit for ease)
3. Open web console for error notation.
4. Navigate to any web page. 
5. Browse the page for the minimum browsing time. 
6. Confirm that the publisher toggle button is enabled (Yellow if site was pinned when included in the ledger, deeper black if merely included in the ledger)
7. In the same tab, navigate to a different web page
8. Repeat steps 5 and 6.
9. Open a new tab, navigate to a new web page
10. Repeat steps 5 and 6


## Reviewer Checklist:


Tests



Given that there didn't seem to be an appropriate appConstant (or a relatively efficient/telling new one that could be made), it seems to make sense for the Publisher Toggle component to have an attribute of its internal state that can check its ledger status at an interval.

Currently, the interval avoids starting until after APP_TOP_SITE_DATA_AVAILABLE has dispatched. Doing this call to pageDataChanged before that action can break ledger activity recorder.

The interval will clear once the publisher toggle has reached an enabled state (there's no way for a site to become disabled merely by browsing, so this interval does not concern that path), any time the publisherToggle component dismounts, and reset if the tab loads to a new potential publisher.
